state/utils/utils.go

```diff
@@ -0,0 +1,17 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package utils
+
+func Marshal(val interface{}, marshaler func(interface{}) ([]byte, error)) ([]byte, error) {
+	var err error
+	bt, ok := val.([]byte)
+	err = nil
```

Comment:
no need to set error to nil. in go, default error value is nil.

if you want to set explicitly, 

I would change line 9.
```
var err error = nil
```